joomla/joomla-cms
# [5.2] Fix Permissions for Manually Running Scheduled Tasks

Pull Request for Issue #36677.

### Summary of Changes
- [x] Explicitly disables the "run task" button on insufficient authorization.
- [x] Adds tooltip to give context on disabling the button.
- [ ] Change default permission level for running tasks/grant access to creator.
- [ ] Prevent manual runs from being publicly accessible regardless of permission levels. Ref: [#36453#issuecomment-1001793962](https://github.com/joomla/joomla-cms/issues/36453#issuecomment-1001793962) / @PhilETaylor 

### Testing Instructions
- Create a new task with an administrator (not superuser) account.
- Still as administrator, try to run it from the Scheduled tasks manager.
- Try to run other tasks created by the super user.
- Login as super user.
- Check that you can run all tasks.
- Thrash some task.
- Check the trashed tasks.

### Actual result BEFORE applying this Pull Request
As administrator (not superuser), there is no indication that the user is not authorized to run the task. The button is usable.
In all cases there is an error message.
Only a super user can run tasks.

### Expected result AFTER applying this Pull Request
As administrator (not superuser):
- The button is enabled for tasks created by yourself and you can run these tasks.
- The button is disabled for tasks which were created by the super user (.e.g those from a new installation). Reason for being disabled is available through a tooltip on hover.

As super user buttons of all tasks are enabled, and you can run these tasks.

When a task is trashed, the button is disabled, and a tool tip shows the reason.

### Documentation Changes Required
N/A
